src/tests/suite/daos_obj.c

```diff
@@ -750,6 +839,118 @@ io_overwrite(void **state)
+	int			rc;
+	daos_size_t		nvme_initial_size;
+	daos_size_t		nvme_current_size;
+	void const *const aggr_disabled[] = {"disaled"};
+	void const *const aggr_set_time[] = {"time"};
+
+	/* choose random object */
+	oid = dts_oid_gen(dts_obj_class, 0, arg->myrank);
+	ioreq_init(&req, arg->coh, oid, DAOS_IOD_ARRAY, arg);
+
+	/* Alloc and set buffer to be a sting*/
+	D_ALLOC(ow_buf, size);
+	assert_non_null(ow_buf);
+	dts_buf_render(ow_buf, size);
+	/* Alloc the fetch buffer */
+	D_ALLOC(fbuf, size);
+	assert_non_null(fbuf);
+	memset(fbuf, 0, size);
+
+	/* Disabled Pool Aggrgation */
+	rc = set_pool_reclaim_strategy(state, aggr_disabled);
+	assert_int_equal(rc, 0);
+
+	/* Get the pool info at the beginning */
+	rc = pool_storage_info(state, &pinfo);
+	assert_int_equal(rc, 0);
+	nvme_initial_size = pinfo.pi_space.ps_space.s_free[1];
+
+	/* Set and verify the full initial string in first transaction */
+	rx_nr = size / OW_IOD_SIZE;
+	/* Insert the initial 4K record which will go through NVMe */
+	insert_single_with_rxnr(dkey, akey, /*idx*/0, ow_buf, OW_IOD_SIZE,
+				rx_nr, DAOS_TX_NONE, &req);
+	/* Lookup the initial 4K record */
+	lookup_single_with_rxnr(dkey, akey, /*idx*/0, fbuf, OW_IOD_SIZE, size,
+				DAOS_TX_NONE, &req);
+	/* Verify the 4K data */
+	assert_memory_equal(ow_buf, fbuf, size);
+
+	/**
+	*Get the pool storage information
+	*/
+	rc = pool_storage_info(state, &pinfo);
+	assert_int_equal(rc, 0);
+	nvme_current_size = pinfo.pi_space.ps_space.s_free[1];
+
+	/**
+	* Verify data written on NVMe by comparing the NVMe Free size.
+	*/
+	if (nvme_current_size > nvme_initial_size - size) {
+		fail_msg("\nNVMe_current_size =%"
+			PRIu64 " > NVMe_initial_size = %"
+			PRIu64 "- written size =%" PRIu64 "",
+			nvme_current_size, nvme_initial_size, size);
+	}
+	nvme_initial_size = pinfo.pi_space.ps_space.s_free[1];
+
+	for (record_set = 0, buf_idx = 0; record_set < 10; record_set++) {
+		buf_idx += 20;
+		memset(fbuf, 0, size);
+		/* Change Two bytes value to original array*/
+		ow_buf[buf_idx + 1] = 48;
+		ow_buf[buf_idx + 2] = 49;
+
+		/* Re-write the same array with modified Two values*/
+		insert_single_with_rxnr(dkey, akey, /*idx*/0, ow_buf,
+			OW_IOD_SIZE, 1, DAOS_TX_NONE, &req);
+
+		/*Read and verify the data with Two updated values*/
+		lookup_single_with_rxnr(dkey, akey, /*idx*/0, fbuf,
+			OW_IOD_SIZE, size, DAOS_TX_NONE, &req);
+		assert_memory_equal(ow_buf, fbuf, size);
+
+		/*Verify the pool size*/
+		rc = pool_storage_info(state, &pinfo);
+		assert_int_equal(rc, 0);
+		nvme_current_size = pinfo.pi_space.ps_space.s_free[1];
+		/**
+		*Data written on SCM so NVMe free size should not change.
+		*/
+		if (nvme_current_size != nvme_initial_size) {
+			fail_msg("NVMe_current_size =%"
+				PRIu64" != NVMe_initial_size %" PRIu64"",
+				nvme_current_size, nvme_initial_size);
+		}
+		nvme_initial_size = pinfo.pi_space.ps_space.s_free[1];
+	}
+
+	/* Enabled Pool Aggrgation */
+	rc = set_pool_reclaim_strategy(state, aggr_set_time);
```

Comment:
Is this just to make sure that aggregation still works at the end of the test? should we be calling pool_storage_info() again to see if the pool SCM extents aggregated?

Reply:
Not actually, We are disabling the aggregation for this tests to get the proper storage sizes for the current tests data. So at the end of the test we are enabling the aggregation back so further tests will be running with default settings.